Gemfile

```diff
@@ -27,3 +27,9 @@ end
 if RUBY_VERSION >= '3.4'
   gem 'mutex_m' # TODO: remove this once as-notifications has such a dependency
 end
+
+if RUBY_VERSION >= '4.0'
+  gem 'pstore'
+  gem 'logger'
+  gem 'ostruct'
+end
```

Comment:
Added to Gemfile instead of gemspec since these are optional features (tracing) and dev concerns (testing), not core to webmachine - no need to bloat the dependency tree for all users.